agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -98,38 +196,80 @@ func getTaskFromRequest(state dockerstate.TaskEngineState, r *http.Request) (*ap
 		logger.Error("Failed to find task ARN for task protection request", logger.Fields{
 			loggerfield.Error: err,
 		})
-		return nil, http.StatusBadRequest, errors.New("Invalid request: no task was found")
+		return nil, http.StatusBadRequest, errors.New("invalid request: no task was found")
 	}
 
 	task, found := state.TaskByArn(taskARN)
 	if !found {
 		logger.Critical("No task was found for taskARN for task protection request", logger.Fields{
 			loggerfield.TaskARN: taskARN,
 		})
-		return nil, http.StatusInternalServerError, errors.New("Failed to find a task for the request")
+		return nil, http.StatusInternalServerError, errors.New("failed to find a task for the request")
 	}
 
 	return task, http.StatusOK, nil
 }
 
 // GetTaskProtectionHandler returns a handler function for GetTaskProtection API
-func GetTaskProtectionHandler(state dockerstate.TaskEngineState,
-	cluster string) func(http.ResponseWriter, *http.Request) {
+func GetTaskProtectionHandler(state dockerstate.TaskEngineState, credentialsManager credentials.Manager,
+	factory TaskProtectionClientFactoryInterface, cluster string) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
-		getTaskProtectionRequestType := "api/v1/GetTaskProtection"
+		getTaskProtectionRequestType := "api/GetTaskProtection/v1"
 
 		task, responseCode, err := getTaskFromRequest(state, r)
 		if err != nil {
 			writeJSONResponse(w, responseCode, err.Error(), getTaskProtectionRequestType)
 			return
 		}
 
-		// TODO: Call ECS
+		ecsClient, responseCode, err := factory.newTaskProtectionClient(credentialsManager, task)
+		if err != nil {
+			writeJSONResponse(w, responseCode, err.Error(), getTaskProtectionRequestType)
+			return
+		}
+
 		logger.Info("GetTaskProtection endpoint was called", logger.Fields{
 			loggerfield.Cluster: cluster,
 			loggerfield.TaskARN: task.Arn,
 		})
-		writeJSONResponse(w, http.StatusOK, "Ok", getTaskProtectionRequestType)
+
+		response, err := ecsClient.GetTaskProtection(&ecs.GetTaskProtectionInput{
+			Cluster: aws.String(cluster),
+			Tasks:   aws.StringSlice([]string{task.Arn}),
+		})
+
+		if err != nil {
+			exceptionType, statusCode := getExceptionTypeAndStatusCode(err)
+			logger.Info(fmt.Sprintf("ECS throws an exception with type %s", exceptionType))
```

Comment:
Should we include `err` in this log? I think it might contain useful info.

Reply:
will do

Reply:
Actually change log level to error would be better I think